### ISSUE TITLE
Create UI to collection user inputs for Self Report Survey

### ISSUE THREAD

**Issue by qingqingliu** [safe-paths-contrib/covid-safe-paths#12]
- [ ] Create UI to match design spec to collection user inputs for self reporting survey
- [ ] Support multi-page UI with back and next support based on design

**Comment by kujenga** [comment]
Update: Setup adding/updated dependencies, working through cross-platform navigation differences. Creating general purpose survey powered by dynamic data.

Not blocked!

**Comment by kujenga** [comment]
Update: basic screens sketched up, hoping to connect with @andrew-sfdc about what a few specifics will look like. Thinking is that it will be more efficient to integrate everything into a single flow.

Question: when and what is being uploaded? Upload is opt-in.

API spec is here: https://docs.google.com/document/d/18laAJLw2usCZvEF7pPF3b_m6IUBr-Vjplz1f7byltTM/edit worked in #6

**Comment by kujenga** [comment]
@aputinski has a few questions for Andrew here that will be followed up on with @andrew-sfdc but not presently blocked/

**Comment by kujenga** [comment]
Update: For the first release the first question and four finish screens will be hard-coded in the app. This allows the API to be simpler.

Will not worry about saving answers, with https://github.com/safe-paths-contrib/covid-safe-paths/issues/19 in the icebox.

**Comment by kujenga** [comment]
Update: All good here, just waiting on UX clarifications: https://github.com/safe-paths-contrib/covid-safe-paths/issues/20#issuecomment-619123307

**Comment by kujenga** [comment]
Update: have the screens that are needed now, will be going back and doing localizations, will be looking at fonts and a few other things.

Outstanding questions:
- There are a few links in the mockups, need to figure out targets and behavior
- What happens after you "share anonymized data", close screen, etc. @andrew-sfdc hoping to connect with @aputinski about what is easiest to implement

**Comment by kujenga** [comment]
Update: Met with @andrew-sfdc and got mockups together for remaining flows. @aputinski and @andrew-sfdc will connect with @qingqingliu about how the captcha will work.

Question flows and ending screens, etc. are working. Things are in a good place.

Possible blockers coming up that Paolo will meet with @aputinski to help work through, will include @imanzano and @qingqingliu as well.

**Comment by kujenga** [comment]
Update: All pushed to a branch, @qingqingliu running it locally. UX sync yesterday and will sync again after this meeting.

Still working on intro screen and @aputinski will work on response payload for the server.

Will then sit with @andrew-sfdc to do final UI tweaks.

**Comment by kujenga** [comment]
Update: all integrated now, work connecting to server started. Test data is not good. Will need more complex branching, etc. to truly validate.

Loading states still remain, spinner while fetch is in progress.

Question: what should we be doing with bad data? can we trust the server? ticket with more detail here: https://github.com/safe-paths-contrib/covid-safe-paths/issues/50

**Comment by kujenga** [comment]
@aputinski to re-group with @qingqingliu on the details of connecting work n https://github.com/safe-paths-contrib/covid-safe-paths/issues/47 to this work.

**Comment by kujenga** [comment]
@aputinski question on timeline for translations: what is the timeline?

Do we want a static version of the survey within the app? And does this need translations as well?

Sam: Translation will need Haitian Creole definitely, Tim Serat can be reached out to for discussing this.

Paolo brings up, should we have the questions themselves localized? Sam and Paolo will work on this.

---

Summary: Use static file with built-in localization capabilities. @aputinski will look into this more.

**Comment by kujenga** [comment]
Update: Converted CDC questions into a hard-coded format that will be added into the app. Also some discussion about translation.

Server will allow an `Accept-Language` header in the future to handle this.

Client will pass this header and it will get back all surveys related to that language.

@aputinski and @ppadmanabhan-sf will connect offline

**Comment by kujenga** [comment]
Update: working on starting and ending screens, maybe need another ending screen.

One pending item, there is still some pending copy that is needed. @aputinski going with what we have today.

@andrew-sfdc will go through that piece and can help fill that role. Ranna can also sync up on this to help out. Maybe be some folks on the SafePaths designer side to help here as well.

**Comment by kujenga** [comment]
Update: lots of cleanup yesterday, will be going through flows and copy today to validate.

Punting for now on full server interactions, some polish needed there, loading states, etc. Can come back to this in the future.

Aiming for wrap up today!

**Comment by kujenga** [comment]
Update: Met with Andrew and Paolo yesterday, got some additional assets.

Text for alert box if you don't "agree" to the first question is TBD, that should be last thing.

Aiming to do a hand-off with Paolo and Ranna, today or tomorrow.

PR will be opened without the server component fully polished. Will be mergeable for the MIT version that is all local.

**Comment by kujenga** [comment]
Update: @andrew-sfdc will get text to @aputinski for completing the last pieces of this.

Adam to get up a PR today with instructions for any future work. PR is intended to be mergable, just without completed server connectivity. It will work with local data.

**Comment by kujenga** [comment]
Update: Tim reviewing code, Ali referring UX.

**Comment by kujenga** [comment]
Update: PR was submitted last week, there is some outstanding feedback.

@aputinski has requirements to go back to his team, would need manager approval to spent some more time. Paolo will reach out accordingly.

Some of the feedback was on the UX side, @aputinski will review with Paolo later today to discuss specific points.

**Comment by kujenga** [comment]
Spent some time yesterday to clean up initial issues, a few more were raised, but those will be handed off to the team reviewing the PR.

Sounds like things are in a good place to be brought in, PR has comments about what needs to be updated.

**Comment by kujenga** [comment]
PR moving forward, folks on the open source project will tackle the last remaining items: https://github.com/Path-Check/covid-safe-paths/pull/776
